lib/Runtime/HostManager/HostManager.cpp

```diff
@@ -78,19 +78,23 @@ llvm::Error HostManager::addNetwork(Module *M) {
   if (backend_) {
     CompilationOptions opts;
     opts.mode = CompilationMode::Infer;
-    for (auto F : M->getFunctions()) {
+    for (auto F : module->getFunctions()) {
       backend_->optimizeFunction(F, opts);
     }
   }
-  auto partitioner = Partitioner(M, deviceInfo);
+  auto partitioner = Partitioner(module.get(), deviceInfo);
   auto nodeList = std::move(partitioner.Partition());
 
-  RETURN_IF_ERR(provisioner_->provision(nodeList, *M));
+  RETURN_IF_ERR(provisioner_->provision(nodeList, *module));
 
   for (auto &node : nodeList) {
-    networks_.emplace((node.root)->name, std::move(node));
+    auto &networkData = networks_[(node.root)->name];
+    networkData.dag = std::move(node);
+    networkData.placeholders = (module->takePlaceholders());
   }
 
+  modules_.push_back(std::move(module));
```

Comment:
Do we need to store the module here? If so when do we free it?

Reply:
Yes good catch, we shouldn't need this now.